### ISSUE TITLE
Ability to localize permissions

### ISSUE THREAD

**Issue by Piedone** [OrchardCMS/OrchardCore#10551]
### Is your feature request related to a problem? Please describe.
Currently, the `Name` and `Description` of `Permission`s are not really suited for localization. You can I suppose return `Permission` objects with `Description` strings freshly run through the localizer in your `IPermissionProvider.GetPermissionsAsync()` implementation but that would be awkward to do every time. And for `Name`, you can't do localization because it's a technical identifier.

### Describe the solution you'd like
Ideally, `Permission` would have `LocalizedString` `DisplayName` and `Description` fields that would work together with the rest of the permission ecosystem.

### Describe alternatives you've considered
You could also run the strings through a localizer when displaying them (currently probably only in Edit.cshtml of `OrchardCore.Roles`). Thus, you could hand-build the necessary PO file.

**Comment by hishamco** [comment]
> Ideally, Permission would have LocalizedString DisplayName and Description fields that would work together with the rest of the permission ecosystem.

I'm not sure why do we need to localization `Name`?!! we always use description in the UI, `Name` is sort of technical ID

**Comment by Piedone** [comment]
That's true. I guess due to this we only ever thought `Description` to be something that should alone convey the meaning of the permission to an end-user. Because of that I'd argue that they're rather short and would better be as names, with `Description` being something that elaborates what the permission actually does more.

So, I'd consider adding a `DisplayName`, use it where `Description` is used currently, and display the new localized `Description` like an additional hint.

**Comment by hishamco** [comment]
I already added a `DisplayName` in a local commit, but I was asking myself where should we display them?

**Comment by Piedone** [comment]
I see, then I'd suggest what I mentioned in my last sentence.

**Comment by hishamco** [comment]
> I see, then I'd suggest what I mentioned in my last sentence.

I think `DisplayName` should be short than description otherwise they will be quite similar, I will push a PR for introducing the `DisplayName` then we can come again to this issue. `DisplayName` is very useful and could be localized

**Comment by sebastienros** [comment]
If we added a display name, what would we add in it, and where would it be displayed? Isn't the description sufficient in the UI?

**Comment by Piedone** [comment]
I'm arguing it isn't, see: https://github.com/OrchardCMS/OrchardCore/issues/10551#issuecomment-955393499

**Comment by hishamco** [comment]
I started a draft PR to fix the issue, if you are agreed with I can apply the same changes everywhere else

**Comment by gvkries** [comment]
This is now possible with the function for dynamic data localization. Please note that there is a small issue that prevents the localization from being displayed, but that should be quickly resolved through #19070.
